src/output.cpp

```diff
@@ -230,4 +291,290 @@ print_overlap_check()
+  }
+}
+
+//==============================================================================
+
+void print_generation()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = overall_generation() - 1;
+  int n = simulation::current_batch > settings::n_inactive ?
+    settings::gen_per_batch*n_realizations + simulation::current_gen : 0;
+
+  // Set format for values
+  std::cout << std::fixed << std::setprecision(5);
+
+  // write out information batch and option independent output
+  std::cout << "  "  << std::setw(9) <<  std::to_string(simulation::current_batch)
+    + "/" + std::to_string(simulation::current_gen) << "   " << std::setw(8)
+    << simulation::k_generation[i];
+
+  // write out entropy info
+  if (settings::entropy_on) {
+    std::cout << "   " << std::setw(8) << simulation::entropy[i];
+  }
+
+  if (n > 1) {
+    std::cout << "   " << std::setw(8) << simulation::keff << " +/-"
+      << std::setw(8) << simulation::keff_std;
+  }
+  std::cout << '\n';
+
+  // Restore state of cout
+  std::cout.flags(f);
+}
+
+//==============================================================================
+
+void print_batch_keff()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = simulation::current_batch*settings::gen_per_batch - 1;
+  int n = n_realizations*settings::gen_per_batch;
+
+  // Set format for values
+  std::cout << std::fixed << std::setprecision(5);
+
+  // write out information batch and option independent output
+  std::cout << "  "  << std::setw(9) <<  std::to_string(simulation::current_batch)
+    + "/" + std::to_string(settings::gen_per_batch) << "   " << std::setw(8)
+    << simulation::k_generation[i];
+
+  // write out entropy info
+  if (settings::entropy_on) {
+    std::cout << "   " << std::setw(8) << simulation::entropy[i];
+  }
+
+  if (n > 1) {
+    std::cout << "   " << std::setw(8) << simulation::keff << " +/-"
+      << std::setw(8) << simulation::keff_std;
+  }
+  std::cout << '\n';
+
+  // Restore state of cout
+  std::cout.flags(f);
+}
+
+//==============================================================================
+
+void show_time(const char* label, double value)
+{
+  std::cout << " " << std::setw(33) << std::left << label << " = "
+    << std::setw(10) << std::right << value << " seconds\n";
+}
+
+void show_rate(const char* label, double value)
+{
+  std::cout << " " << std::setw(33) << std::left << label << " = " <<
+    value << " particles/second\n";
+}
+
+void print_runtime()
+{
+  using namespace simulation;
+
+  // display header block
+  header("Timing Statistics", 6);
+
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // display time elapsed for various sections
+  std::cout << std::scientific << std::setprecision(4);
+  show_time("Total time for initialization", time_initialize.elapsed());
```

Comment:
Based on how it's used here, maybe `show_time` should take an indent level?

Reply:
Good suggestion, I'll add that